### PR TITLE
Override ShaderGUI per ShaderGraph [8.x.x Backport][skip ci]

### PR DESCRIPTION
**Pull request by zadi0648**
Backport of #6192 

**Because the Backports don't deal with Targets, there are these changes:**

* This version has no CurrentPipelinePreferredShaderGUI, as that's only achievable with Targets. As such things work slightly different, where the TextField doesn't get automatically filled the pipeline's default once OverrideEnabled is toggled on.
* See comments for Backport Exclusive changes in the code.

---
**Yamato:**
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/8.x.x%252Fsg%252Fshadergui-override-per-sg

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
>At this moment, this version only has support for PBR & Unlit
is it plan to add other HDRP master node?

**Comment by zadi0648** [comment]
yes.

File: com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubShader.cs [hunk 1]
```diff
@@ -547,7 +547,12 @@ public string GetSubshader(IMasterNode iMasterNode, GenerationMode mode, List<st
             }
             subShader.Deindent();
             subShader.AddShaderChunk("}", true);
-            subShader.AddShaderChunk(@"CustomEditor ""UnityEditor.Rendering.HighDefinition.DecalGUI""");
+
+            if (!masterNode.OverrideEnabled)
+            {
+                subShader.AddShaderChunk(@"CustomEditor ""UnityEditor.Rendering.HighDefinition.DecalGUI""");
+            }
```
Comment (zadi0648) [inline com.unity.render-pipelines.high-definition/Editor/Material/Decal/ShaderGraph/DecalSubShader.cs]: Backport Exclusive: all of the applicable SubShaders have this check now.

File: com.unity.shadergraph/Editor/Data/Nodes/MaterialMasterNode.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+using UnityEditor.Graphing;
```
Comment (zadi0648) [inline com.unity.shadergraph/Editor/Data/Nodes/MaterialMasterNode.cs]: Backport Exclusive: This class is only in the Backports (and makes things easier)

File: com.unity.shadergraph/Editor/Drawing/Views/MasterNodeSettingsView.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            // Display the ShaderGUI text field only when the override is enabled
+            if (m_CanChangeShaderGUI.OverrideEnabled)
+            {
+                m_PropertySheet.Add(m_OverrideFieldRow);
+
+                ProcessShaderGUIField(storedValue, false);
+            }
+            else if (m_PropertySheet.Contains(m_OverrideFieldRow))
+            {
+                m_PropertySheet.Remove(m_OverrideFieldRow);
```
Comment (zadi0648) [inline com.unity.shadergraph/Editor/Drawing/Views/MasterNodeSettingsView.cs]: Backport Exclusive: We cannot support the Field automatically containing the pipeline's default when OverrideEnabled is enabled due to the lack of targets

**Pull request merged**